### PR TITLE
Changed all the deprecated armadillo matrix initializations with the new braced initializer list

### PR DESCRIPTION
**Pull request by RishabhGarg108**
Changed the matrix initializations throughout the mlpack codebase as discussed in #2699 
There were a lot of files to be modified and it was practically difficult to do everything manually. So, I would like to give a short brief on what I did.

1.  I did `grep -Rw -n ./ -e 'arma::endr\|endr'` inside the `src/mlpack` folder. This gave me all those files and line that used the deprecated initialization.
2.  Then I wrote a small python regular expression script that replaced `<<` with a `,`.
3.  Then after that I manually fixed the spacing and indentation.
4.  Finally I did `grep -Rw -n ./ -e '<<'`. To see if I missed some column initializations and then fixed them.

So, I believe that I have fixed all the initializations in the codebase.

I have left two files in the above process. One of them is done in #2705 while the other one was having some issue as mentioned [here](https://github.com/mlpack/mlpack/issues/2699#issuecomment-723965799).

### CODE REVIEW

**Comment by RishabhGarg108** [comment]
@conradsnicta , can you please invest some time to review it. Thanks! :)

**Comment by zoq** [comment]
Can you merge in the latest master branch.

**Comment by RishabhGarg108** [comment]
@zoq , I made the requested changes, and also updated the one remaining file that was causing error.
Can you please review it. Thanks!

File: src/mlpack/tests/binarize_test.cpp [hunk 1]
```diff
@@ -46,9 +46,9 @@ TEST_CASE("BinarizeOneDimension", "[BinarizeTest]")
 TEST_CASE("BinerizeAll", "[BinarizeTest]")
 {
   mat input;
-  input << 1 << 2 << 3 << endr
-        << 4 << 5 << 6 << endr // this row will be tested
-        << 7 << 8 << 9;
+  input = { { 1, 2, 3 },
+            { 4, 5, 6 }, // this row will be tested
```
Comment (zoq) [inline src/mlpack/tests/binarize_test.cpp:50]:
```suggestion
            { 4, 5, 6 }, // This row will be tested.
```

I know the comment was there before, but we should be consistent and use the same comment style.

File: src/mlpack/tests/lsh_test.cpp [hunk 2]
```diff
@@ -689,12 +689,11 @@ BOOST_AUTO_TEST_CASE(RecallTestPartiallyCorrect)
...
-    2 << arma::endr <<
-    3 << arma::endr <<
-    4 << arma::endr <<
-    6 << arma::endr <<
-    7 << arma::endr;
+  q2 << 2 << arma::endr
+     << 3 << arma::endr
+     << 4 << arma::endr
+     << 6 << arma::endr
+     << 7 << arma::endr;
```
Comment (zoq) [inline src/mlpack/tests/lsh_test.cpp:696]: This needs to be adapted as well.

File: src/mlpack/tests/maximal_inputs_test.cpp [hunk 3]
```diff
@@ -73,12 +71,12 @@ BOOST_AUTO_TEST_CASE(ColumnToBlocksChangeBlockSize)
...
-               << -3 << -1 << -0.71429 << -0.42857 << -0.14286
-               << -3 << 0.14286 << 0.42857 << 0.71429 << 1 << -3 << arma::endr
-               << -3 << -3 << -3 << -3 << -3 << -3 << -3 << -3 << -3 << -3
-                  << -3 << arma::endr;
+  matlabResults = { { -3, -3,       -3,       -3,       -3, -3,      -3,
+                          -3,      -3, -3, -3 },
+                    { -3, -1, -0.71429, -0.42857, -0.14286, -3, 0.14286,
+                     0.42857, 0.71429,  1, -3 },
+                    { -3, -3,       -3,       -3,       -3, -3,      -3,
+                          -3,      -3, -3, -3 } };
```
Comment (zoq) [inline src/mlpack/tests/maximal_inputs_test.cpp:79]: This looks strange, I guess in this case no need to align the values if that allows us to put the values on the same line.

**Comment by RishabhGarg108** [comment]
Hey @zoq , I made the requested changes. Can you have a look again?

**Review by zoq — APPROVED** [review]
Looks great to me, thanks for putting this together.

**Comment by RishabhGarg108** [comment]
Always happy to contribute :D
Learning tons of new things while working on these.

File: src/mlpack/tests/lsh_test.cpp [hunk 4]
```diff
@@ -132,8 +132,8 @@ BOOST_AUTO_TEST_CASE(NumTablesTest)
     fail = false;
 
     const int lSize = 6; // Number of runs.
-    const int lValue[] = {1, 8, 16, 32, 64, 128}; // Number of tables.
-    double lValueRecall[lSize] = {0.0}; // Recall of each LSH run.
+    const int lValue[] = { 1, 8, 16, 32, 64, 128 }; // Number of tables.
```
Comment (shrit) [inline src/mlpack/tests/lsh_test.cpp:135]: @zoq Are we following this spacing style for non-arma matrices

**Review by shrit — APPROVED** [review]

File: src/mlpack/tests/lsh_test.cpp (hunk 4 above)
Comment (zoq) [reply src/mlpack/tests/lsh_test.cpp:135]: Hm, I think we don't have a style guide for such cases, maybe we do?

Comment (shrit) [reply src/mlpack/tests/lsh_test.cpp:135]: I do not know either, I usually put spaces, but my style is different from the one used in mlpack, so I am checking

**Pull request merged**